class-tgm-plugin-activation.php

```diff
@@ -719,6 +693,54 @@ public function maybe_adjust_source_dir( $source, $remote_source, $upgrader ) {
 
 			return $source;
 		}
+		
+		/**
+		 * Activate a single plugin and send feedback about the result to the screen.
+		 *
+		 * @since 2.5.0
+		 *
+		 * @param string $file_path Path within wp-plugins/ to main plugin file
+		 * @param string $slug      Plugin slug
+		 * @param bool   $automatic Whether this is an automatic activation after an install. Defaults to false.
+		 *                          This determines the styling of the output messages.
+		 *
+		 * @return bool True if an error was encountered, false otherwise.
+		 */
+		protected function activate_single_plugin( $file_path, $slug, $automatic = false ) {
+
+			if ( ! is_plugin_active( $file_path ) ) {
+				$activate = activate_plugin( $file_path ); // Activate the plugin.
+
+				if ( is_wp_error( $activate ) ) {
+					echo '<div id="message" class="error"><p>', wp_kses_post( $activate->get_error_message() ), '</p></div>',
+						'<p><a href="', esc_url( add_query_arg( 'page', urlencode( $this->menu ), self_admin_url( $this->parent_slug ) ) ), '" target="_parent">', esc_html( $this->strings['return'] ), '</a></p>';
+					return true; // End it here if there is an error with activation.
+
+				} else {
+					if ( ! $automatic ) {
+						// Make sure message doesn't display again if bulk activation is performed immediately after a single activation.
+						if ( ! isset( $_POST['action'] ) ) {
+							echo '<div id="message" class="updated"><p>', esc_html( $this->strings['activated_successfully'] ), ' <strong>', esc_html( $this->plugins[ $slug ]['name'] ), '.</strong></p></div>';
+						}
+					}
+					else {
+						// Simpler message layout for use on the plugin install page
+						echo '<p>', esc_html( $this->strings['plugin_activated'] ), '</p>';
```

Comment:
Ouch, no - massive overhead which we can do without
See: https://core.trac.wordpress.org/browser/tags/4.2.1/src/wp-includes/formatting.php#L373